src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.FeatureCollection.cs

```diff
@@ -73,5 +75,11 @@ async Task IHttpResponseCompletionFeature.CompleteAsync()
                 await ProduceEnd();
             }
         }
+
+        void IHttpResetFeature.Reset(int errorCode)
+        {
+            var abortReason = new ConnectionAbortedException(CoreStrings.FormatHttp2StreamResetByApplication((Http2ErrorCode)errorCode));
```

Comment:
Do we want to validate the errorCode is a valid Http2ErrorCode?

Reply:
No, they can add new ones at any time. Also the server (and in most cases the client) does not behave any differently based on the value.